server/child-processes/create-bundle.js

```diff
@@ -0,0 +1,217 @@
+const path = require( 'path' );
+const sander = require( 'sander' );
+const child_process = require( 'child_process' );
+const targz = require( 'tar.gz' );
+const request = require( 'request' );
+const browserify = require( 'browserify' );
+const rollup = require( 'rollup' );
+const resolve = require( 'rollup-plugin-node-resolve' );
+const UglifyJS = require( 'uglify-js' );
+const isModule = require( 'is-module' );
+const makeLegalIdentifier = require( '../utils/makeLegalIdentifier' );
+
+const { root, tmpdir } = require( '../../config.js' );
+
+process.on( 'message', message => {
+	if ( message.type === 'start' ) {
+		createBundle( message.params );
+	}
+});
+
+process.send( 'ready' );
+
+async function createBundle ({ hash, pkg, version, deep, query }) {
+	const dir = `${tmpdir}/${hash}`;
+	const cwd = `${dir}/package`;
+
+	try {
+		await sander.mkdir( dir );
+		await fetchAndExtract( pkg, version, dir );
+		await sanitizePkg( cwd );
+		await installDependencies( cwd );
+		
+		const code = await bundle( cwd, deep, query );
+
+		info( `[${pkg.name}] minifying` );
+
+		let result;
+
+		try {
+			result = UglifyJS.minify( code ).code;
+		} catch ( err ) {
+			info( `[${pkg.name}] minification failed: ${err.message}` );
+			result = code;
```

Comment:
@Rich-Harris `uglify-js@3` has a new API. It will return `result.error` in the event of an error.

https://github.com/mishoo/UglifyJS2/blame/v3.0.26/README.md#L470-L474